wpilibj/src/main/java/edu/wpi/first/wpilibj/TimesliceRobot.java

```diff
@@ -107,7 +111,14 @@ public void schedule(Runnable func, double allocation) {
               + "\n");
     }
 
-    addPeriodic(func, m_controllerPeriod, m_nextOffset);
+    var fmt = new DecimalFormat("#0.00");
+    String name =
+        "Timeslice{"
+            + new DecimalFormat("#0.00").format(m_nextOffset)
+            + "->"
+            + fmt.format(m_nextOffset + allocation)
+            + "}";
```

Comment:
The name should be as short as possible, also I don't think performance is very important here because this method will be called a handful of times at bootup at most.

Reply:
> The name should be as short as possible

True to an extent, but I think "as short as possible" is 3 digits of precision (without the unit indicator or spaces). For example, using the example in the [`TimesliceRobot` class documentation](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/TimesliceRobot.html), we could end up with 3 different periodic methods that are all named `Timeslice{0.00->0.00}`, which would cause issues with name collisions. If we instead had 3 digits of precision, we would have `Timeslice{0.002->0.004}`, `Timeslice{0.004->0.004}`, `Timeslice{0.004->0.005}`, which is a lot better.

> I don't think performance is very important here because this method will be called a handful of times at bootup at most

Fair, but for the sake of code readability, we could at least either reuse the `DecimalFormat` variable or create them inline both times.